### DISCUSSION TITLE
Merkle Patricia Forestry Questions

### DISCUSSION THREAD

**Discussion by AgustinBadi** [aiken-lang/merkle-patricia-forestry#3] [General]
Hey! I have just finished to check the repo and it is really interesting, the reduction of neighbourhoods at each step was a very clever solution.

Still there is a few things that I don't fully grasp: 

1. The key is the digest of the value? Or the relation between the key and the value is arbitrary? 

2. Why at the benchmark table the tries size are exponents of 10? Wouldn't must be instead exponents of 16?

Ty!

**Comment by KtorZ** [comment]
Hey sorry, I completely missed the question. Somehow didn't get a notification. A late reply still for the record:

1. The key isn't necessarily the digest of the value (can be, but not required). The key can literally be _anything_. A (merkle) patricia trie is analogous to a key:value map. Whereas a merkle tree is analogous to a list.  

2. I used exponent of 10 because as humans, we reason better with powers of 10. In the end, the relation is logarithmic, and involves base-2 logs. What scale we then use to showcase this is flexible / open. What matters is seing that the cost are not linear in the size.
